main_test.go

```diff
@@ -41,10 +50,84 @@ func TestCafeWhenOk(t *testing.T) {
 	}
 	for _, v := range requests {
 		response := httptest.NewRecorder()
+        
 		req := httptest.NewRequest("GET", v, nil)
 
 		handler.ServeHTTP(response, req)
 
 		assert.Equal(t, http.StatusOK, response.Code)
 	}
 }
+
+func TestCafeCount(t *testing.T) {
+    handler := http.HandlerFunc(mainHandle)
+
+    requests := []struct {
+        count int // передаваемое значение count
+        want  int // ожидаемое количество кафе в ответе
+    }{
+        {0, 0},
+        {1, 1},
+        {2, 2},
+        {100, len(cafeList["moscow"])}, // Москва имеет 5 кафе
```

Comment:
Можно так на случай если в Москве будет больше 100 кафе
```suggestion
        {100, min(len(cafeList[city]), 100)}, // Москва имеет 5 кафе
```

Reply:
А понял, не подумал насчёт масштабируемости.
Спасибо!